### PR TITLE
feat: support koa2

### PR DESCRIPTION
**Pull request by ngot**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -125,68 +125,67 @@ module.exports = function (options) {
...
-    if (!this.session) {
-      debug('this.session not exists');
+    if (!ctx.session) {
+      debug('ctx.session not exists');
       // ignore not match path
       if (!loginRequired) {
-        debug('not match needLogin path, %j', this.path);
-        return yield next;
+        debug('not match needLogin path, %j', ctx.path);
+        return await next();
```
Comment (fengmk2) [inline index.js]: return next(), no need to await it.

File: index.js [hunk 2]
```diff
@@ -125,68 +125,67 @@ module.exports = function (options) {
...
       }
       debug('relogin again');
-      return yield loginHandler.call(this, next);
+      return await loginHandler.call(ctx, next);
```
Comment (fengmk2) [inline index.js]: `return await loginHandler(ctx, next);` don't binding `this` from now on.

File: index.js [hunk 3]
```diff
@@ -125,68 +125,67 @@ module.exports = function (options) {
...
     }
 
     // get login path
-    if (this.path === options.loginPath) {
+    if (ctx.path === options.loginPath) {
       debug('match login path');
-      return yield loginHandler.call(this, next);
+      return await loginHandler.call(ctx, next);
```
Comment (fengmk2) [inline index.js]: too

File: index.js [hunk 4]
```diff
@@ -125,68 +125,67 @@ module.exports = function (options) {
...
     }
 
     // get login callback
-    if (this.path === options.loginCallbackPath) {
+    if (ctx.path === options.loginCallbackPath) {
       debug('match login clalback path');
-      return yield loginCallbackHandler.call(this, next);
+      return await loginCallbackHandler.call(ctx, next);
```
Comment (fengmk2) [inline index.js]: loginCallbackHandler(ctx, next)

**Comment by ngot** [comment]
@fengmk2 done

**Comment by ngot** [comment]
@fengmk2 anyone can review this?

**Comment by denghongcai** [comment]
+ koa-userauth@2.0.0

**Pull request merged**